### DISCUSSION TITLE
Purple screen appearing with streamlink?

### DISCUSSION THREAD

**Discussion by bennyhillthebest** [streamlink/streamlink#4105] [Q&A] [answered]
Hello, today i experienced a couple of times the "purple screen" with streamlink. Is this only a problem of mine, or is it widespread? Thanks for the amazing software!!!

Here is the output:
[stream.hls][debug] Adding segment 320 to queue
[stream.hls][debug] Segment 319 complete
[stream.hls][debug] Segment 320 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 321 to queue
[stream.hls][info] Filtering out segments and pausing stream output
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 322 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 323 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 324 to queue
[stream.hls][debug] Adding segment 325 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 326 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 327 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 328 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 329 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 330 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 331 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 332 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 333 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 334 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 335 to queue
[stream.hls][debug] Adding segment 336 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 337 to queue
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 338 to queue
[stream.hls][debug] Segment 337 complete
[stream.hls][info] Resuming stream output
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Adding segment 339 to queue

**Comment by bastimeyer** [comment, marked as answer]
https://github.com/streamlink/streamlink/issues/3210
Twitch has, as expected, made new changes after their source code has been leaked a few weeks ago.

New access token request headers were added in #4086, but this, as expected as well, stopped working too, at least for non-preroll ads.

The `--twitch-disable-ads` parameter still seems to be able to filter out ads, but there's one HLS segment with the purple screen which doesn't get caught by it, so the purple screen appears before the stream output stops for filtering out the ads. It's possible that some timestamps are set differently now or that they are using different values for annotating the ad segments. Without seeing the actual HLS playlist contents, this can't be fixed.

**Comment by bastimeyer** [comment]
I'm turning this discussion post into an issue thread now, so that this plugin issue can be properly tracked and fixed, and so that people can actually see it in the issue tracker, because there will be lots of duplicates otherwise. Please don't open discussions for actual issues next time, thanks.

edit:
Discussions can't be turned into issues like the other way around, so I'm opening a new issue instead. I will lock this discussion here now.

New plugin issue here:
https://github.com/streamlink/streamlink/issues/4106
